pkgs/applications/editors/emacs/elisp-packages/manual-packages/cask/package.nix

```diff
@@ -1,30 +1,48 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, bash
-, emacs
-, python3
+{
+  lib,
+  ansi,
+  coreutils,
+  dash,
+  ecukes,
+  el-mock,
+  ert-async,
+  ert-runner,
+  f,
+  fetchFromGitHub,
+  git,
+  gitUpdater,
+  melpaBuild,
+  noflet,
+  package-build,
+  s,
+  servant,
+  shell-split-string,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
-  pname = "cask";
+let
   version = "0.8.8";
+in
+melpaBuild {
+  pname = "cask";
+  inherit version;
 
   src = fetchFromGitHub {
+    pname = "cask-source";
+    inherit version;
     owner = "cask";
     repo = "cask";
-    rev = "v${finalAttrs.version}";
+    rev = "v${version}";
     hash = "sha256-TlReq5sLVJj+pXmJSnepKQkNEWVhnh30iq4egM1HJMU=";
   };
 
-  doCheck = true;
+  patches = [
+    substituteAll {
+      source = ./0000-cask-clidir.diff;
+      CASK_CLIDIR = "$out/share/emacs/site-lisp/elpa/cask-${version}";
+    }
```

Comment:
Presumably, `replaceVars` is better: https://github.com/NixOS/nixpkgs/pull/301350

To actually substitute `$out`, we have to do so during build.  For example, adding the following to `postInstall`:

```bash
substituteInPlace $sourceRoot/bin/cask --replace-fail \
  @cask_clidir@ $out/share/emacs/site-lisp/elpa/*-*
```

And replace this `substituteAll` with just the patch itself.

Reply:
I think `substituteAllInPlace` is more akin to what I want to do here.